src/Psy/Plugin/AbstractPlugin.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Psy\Plugin;
+
+abstract class AbstractPlugin
+{
+    public static function register()
+    {
+        Manager::register(new static(), static::getName());
+    }
+
+    /**
+     * @return string
+     */
+    final public static function getName()
```

Comment:
Then @Markcial's register method in the `AbstractPlugin` is wrong. Why instanciate the plugin when all it should do is just a  `__CLASS__` instead ?

(Still subjective, but I really don't like static if I can help otherwise :D)